gateways/paypalpayflow/paypalpayflow.go

```diff
@@ -0,0 +1,197 @@
+	return &PaypalPayflowClient{
+		httpClient: httpClient,
+		partner:    partner,
+		password:   password,
+		vendor:     vendor,
+		user:       user,
+		url:        paypalURL(environment),
+	}
+}
+
+func paypalURL(env common.Environment) string {
+	if env != common.Production {
+		return "https://pilot-payflowpro.paypal.com"
+	}
+	return "https://payflowpro.paypal.com'"
+}
+
+type Request struct {
+	TrxType            string
+	Amount             *string
+	Verbosity          *string
+	Tender             *string
+	CreditCardNumber   *string
+	CardExpirationDate *string
+	OriginalID         *string
+}
+
+type Response map[string]string
+
+func (client *PaypalPayflowClient) sendRequest(request *Request) (*Response, error) {
+	data := fmt.Sprintf(
+		"PARTNER[%d]=%s&PWD[%d]=%s&VENDOR[%d]=%s&USER[%d]=%s&TRXTYPE[%d]=%s",
+		len(client.partner), client.partner,
+		len(client.password), client.password,
+		len(client.vendor), client.vendor,
+		len(client.user), client.user,
+		len(request.TrxType), request.TrxType,
+	)
+
+	if request.Amount != nil {
+		data = data + fmt.Sprintf("&AMT[%d]=%s", len(*request.Amount), *request.Amount)
+	}
+	if request.Verbosity != nil {
+		data = data + fmt.Sprintf("&VERBOSITY[%d]=%s", len(*request.Verbosity), *request.Verbosity)
+	}
+	if request.Tender != nil {
+		data = data + fmt.Sprintf("&TENDER[%d]=%s", len(*request.Tender), *request.Tender)
+	}
+	if request.CreditCardNumber != nil {
+		data = data + fmt.Sprintf("&ACCT[%d]=%s", len(*request.CreditCardNumber), *request.CreditCardNumber)
+	}
+	if request.CardExpirationDate != nil {
+		data = data + fmt.Sprintf("&EXPDATE[%d]=%s", len(*request.CardExpirationDate), *request.CardExpirationDate)
+	}
+	if request.OriginalID != nil {
+		data = data + fmt.Sprintf("&ORIGID[%d]=%s", len(*request.OriginalID), *request.OriginalID)
+	}
+
+	req, err := http.NewRequest("POST", client.url, strings.NewReader(data))
+	if err != nil {
+		log.Fatal(err)
+	}
+	// req.Header.Set("Content-Type", "application/x-www-form-urlencoded")
+	resp, err := client.httpClient.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+	bodyText, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	response := make(Response)
+	for _, line := range strings.Split(string(bodyText), "&") {
+		line := strings.Split(strings.TrimSpace(line), "=")
+		if len(line) != 2 {
+			continue
+		}
+		response[line[0]] = line[1]
+	}
+
+	return &response, nil
+}
+
+// Authorize a transaction. This transaction must be captured to receive funds
+func (client *PaypalPayflowClient) Authorize(request *sleet.AuthorizationRequest) (*sleet.AuthorizationResponse, error) {
+	response, err := client.sendRequest(buildAuthorizeParams(request))
+	if err != nil {
+		return nil, err
+	}
+
+	transactionID, ok1 := (*response)["PNREF"]
+	result, ok2 := (*response)["RESULT"]
+	if ok1 && ok2 && result == "0" {
```

Comment:
Same here, let's make this a const so we can make this more readable. Something like `result == successResponse`

Reply:
OK